apps/007_ofsted_market_analysis/app.py

```diff
@@ -2,113 +2,498 @@
+st.title('Ofsted Market Analysis')
+st.write('This tool analyses a list of social care settings providers as provided by Ofsted.')
+st.write('Use the sidebar selectors to filter by geography and provider type.')
+st.write('Click on the tabs below to view different breakdowns for the selected filters.')
 
 # Takes "Ofsted Social care providers list for LAs"
-uploaded_files = st.file_uploader('Upload Ofsted social care providers list:', accept_multiple_files=False)
+uploaded_files = st.file_uploader('Upload Ofsted social care providers list:', accept_multiple_files=True)
+
 
 if uploaded_files:
-    if uploaded_files.name[-4:] == '.csv':
-        loaded_files = {uploaded_files.name: pd.read_csv(uploaded_files)}
-    else:
-        loaded_files = {uploaded_files.name: pd.read_excel(uploaded_files)}
-    
-    for file in loaded_files.items():
-        st.write('Displaying "' + file[0] + '"') # Print the key from the dictionary for this file
-        #st.dataframe(file[1]) # Print the dataframe from the dictionary for this file
 
-    # Access the dataframe for the file
-    df = list(loaded_files.values())[0]
+    loaded_files = {uploaded_file.name: pd.read_csv(uploaded_file) for uploaded_file in uploaded_files}
+
+    files_dict = {}
+
+    for name,file in loaded_files.items():
+        file.dropna(axis=1, how="all", inplace=True) # Drop any fields with only nan
+        file.dropna(axis=0, how="all", inplace=True) # Drop any rows with only nan
+        year = name[-8:-4] # Find year of dataset from file name
+        file["year"] = year # Place year into new field
+        #st.write(year)
+
+        # Find month of dataset from file name
+        if "april" in name.lower():
+            month = "april"
+            # st.write(month)
+        if "may" in name.lower():
+            month = "may"
+        if "june" in name.lower():
+            month = "june"
+        if "july" in name.lower():
+            month = "july"
+        if "august" in name.lower():
+            month = "august"
+        if "september" in name.lower():
+            month = "september"
+        if "october" in name.lower():
+            month = "october"
+        if "november" in name.lower():
+            month = "november"
+        if "december" in name.lower():
+            month = "december"
+        if "january" in name.lower():
+            month = "january"
+        if "february" in name.lower():
+            month = "february"
+        if "march" in name.lower():
+            month = "march"
+        file["month"] = month  # Populate new field with month name
+
+        # Find CH, Non-CH or all from file name
+        if " non" in name.lower():
+            homestype = "non-ch"
+        elif " ch" in name.lower():
+            homestype = "ch"
+        else:
+            homestype = "all"
+        file["homestype"] = homestype # Place homes type into new field
+
+        name = f"{month}{year}{homestype}"
+
+        # Rename columns to standard names 

```

Comment:
We should be able to do this as one dicitonary with one file.rename() call rather than one per line